rust/fory-derive/src/object/write.rs

```diff
@@ -305,13 +306,13 @@ fn gen_write_field_impl(
     }
 }
 
-pub fn gen_write_data(fields: &[&Field]) -> TokenStream {
-    let write_fields_ts: Vec<_> = get_filtered_fields_iter(fields)
-        .enumerate()
-        .map(|(idx, field)| gen_write_field_with_index(field, idx, true))
+pub fn gen_write_data(sorted_fields: &[SortedField<'_>]) -> TokenStream {
+    let fields: Vec<&Field> = sorted_fields.iter().map(|sf| sf.field).collect();
+    let write_fields_ts: Vec<_> = get_filtered_sorted_fields_iter(sorted_fields)
```

Comment:
TBH, the `sorted` in this PR looks a little vargue to me. We are not sorting here, we jsut assign an alias to every field of the tuple struct

Reply:
Yup ,as I said above, what I want to express is fields with original index